src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CreationPolicy.cs

```diff
@@ -16,6 +16,12 @@ private enum GeneratedDocumentCreationPolicy
         /// </summary>
         Create,
 
+        /// <summary>
+        /// Source generators that are considered required should be run and produce results. Previously
+        /// computed results should be reused for other generators.
+        /// </summary>
+        CreateRequired,
+
         /// <summary>
         /// Source generators should not run.  Whatever results were previously computed should be reused.
         /// </summary>
```

Comment:
why not just do 'DoNotCreate_ExceptRequired' (for the value below).  i'm not sure we want/need 3 values.  just that the second value shoudl always run razor(if you're saying it is always necessary for razor to run).

Reply:
So that's what I was originally going to do. But when running in a debugger it seemed like the `DoNotCreate` path was always run _after_ the create path, even in Automatic mode. That suggested to me that it was still going to be required? In theory we could re-run razor and it should just be a no-op? 

I do not know/understand the intricacies of when this code gets called to know well enough if that would still be acceptable :)